sdk/devtestlabs/Azure.ResourceManager.DevTestLabs/tests/Scenario/DevTestLabVirtualNetworkTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Resources;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.DevTestLabs.Tests
+{
+    internal class DevTestLabVirtualNetworkTests : DevTestLabsManagementTestBase
+    {
+        private DevTestLabVirtualNetworkCollection _dtlVnetCollection;
+        private const string _dtlVnetPrefixName = "dtlVnet";
+
+        public DevTestLabVirtualNetworkTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [SetUp]
+        public async Task SetUp()
+        {
+            TestResourceGroup = await CreateResourceGroup();
+            var lab = await CreateDevTestLab(TestResourceGroup, Recording.GenerateAssetName("lab"));
+            _dtlVnetCollection = lab.GetDevTestLabVirtualNetworks();
+        }
+
+        private async Task<DevTestLabVirtualNetworkResource> CreateDevTestLabVnet(string vnetName)
+        {
+            var data = new DevTestLabVirtualNetworkData(DefaultLocation);
+            var dtlVnet = await _dtlVnetCollection.CreateOrUpdateAsync(WaitUntil.Completed, vnetName, data);
+            return dtlVnet.Value;
+        }
+
+        [RecordedTest]
+        public async Task CreateOrUpdateExistGetGetAllDelete()
+        {
+            // CreateOrUpdate
+            string dtlVnetName = Recording.GenerateAssetName(_dtlVnetPrefixName);
+            var dtlVnet = await CreateDevTestLabVnet(dtlVnetName);
+            ValidateDevTestLabVirtualNetwork(dtlVnet.Data, dtlVnetName);
+
+            // Exist
```

Comment:
consider splitting and put creation into setup, the test name is strange as well.